### PR TITLE
Connect to beacon chain

### LINKED ISSUE #148: Support connecting to network
- implement container.ts so it can run prysmatic beacon node in docker
   - should provide options to connect to various networks (just public for now)
- ability to stream logs
- add redux functions that should run implemented container in background (another thread?!)
- add redux functions to stop and restart that container

### PR DESCRIPTION
**Pull request by morrigan**
Resolves #148

### CODE REVIEW

File: test/e2e/services/docker/chain.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    let beaconChain: BeaconChain;
+
+    // clean up created container
+    afterEach(async () => {
+        if (await BeaconChain.isDockerInstalled()) {
+            const isRunning = await beaconChain.isRunning();
+            if (isRunning) {
+                await beaconChain.stop();
+            }
+            await runCmdAsync("docker rm Prysm-beacon-node").catch();
```
Comment (mpetrunic) [inline test/e2e/services/docker/chain.spec.ts]: use SupportedNetwork.Prysm so this doesn't break if we change

File: test/e2e/services/docker/chain.spec.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            await runCmdAsync("docker volume rm Prysm-chain-data").catch();
```
Comment (mpetrunic) [inline test/e2e/services/docker/chain.spec.ts]: how does await and .catch() works?

File: src/renderer/services/docker/chain.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+type LogType = "info" | "error";
+type LogCallbackFunc = (type: LogType, message: string) => void;
+
+export enum SupportedNetworks {
+    PRYSM = "Prysm",
+}
+
+export class BeaconChain extends Container {
+    public static instance: BeaconChain;
```
Comment (mpetrunic) [inline src/renderer/services/docker/chain.ts]: Hmm i think better solution is to have singleton DockerContainers where each "Container" class could register itself.
In that way you could generate multiple BeaconChain containers like for local and public testnet at the same time

File: src/renderer/services/docker/chain.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public static async startPrysmBeaconChain(waitUntilReady = false): Promise<BeaconChain> {
+        const bc = new BeaconChain({
+            image: "gcr.io/prysmaticlabs/prysm/beacon-chain:latest",
+            name: `${SupportedNetworks.PRYSM}-beacon-node`,
+            restart: "unless-stopped",
+            ports: ["4000:4000", "13000:13000"],
```
Comment (mpetrunic) [inline src/renderer/services/docker/chain.ts]: I don't think port 13000 needs to be exposed to host.
Also user should be able to configure on which port does he wan't port 4000 to be exposed(aka this should be configurable by user)

**Comment by morrigan** [comment]
@mpetrunic pls

**Pull request merged**